app/javascript/packs/MainApp/components/RefereeProfile.jsx

```diff
@@ -7,50 +7,151 @@ class RefereeProfile extends Component {
     match: PropTypes.shape({
       params: PropTypes.object
     }).isRequired
-  }
+  };
 
   state = {
+    httpStatus: 200,
+    httpStatusText: '',
     firstName: '',
     lastName: '',
     bio: '',
-    email: ''
-  }
+    email: '',
+    pronouns: '',
+    nationalGoverningBodies: [],
+    certifications: []
+  };
 
   componentDidMount() {
-    const { match: { params } } = this.props
+    const { match: { params } } = this.props;
 
     axios.get(`/api/v1/referees/${params.id}`)
-      .then(({ data }) => {
-        const { attributes } = data.data
+      .then(({ status, statusText, data }) => {
+        const { data: { attributes }, included } = data;
+        const certifications = included
+          .filter(({ type }) => type === 'certification')
+          .map(certification => certification.attributes);
+        const nationalGoverningBodies = included
+          .filter(({ type }) => type === 'national_governing_body')
+          .map(nationalGoverningBody => nationalGoverningBody.attributes);
         this.setState({
+          httpStatus: status,
+          httpStatusText: statusText,
           firstName: attributes.first_name,
           lastName: attributes.last_name,
           bio: attributes.bio,
-          email: attributes.email
+          email: attributes.email,
+          pronouns: attributes.pronouns,
+          nationalGoverningBodies,
+          certifications
         })
       })
+      .catch(({ response: { status, statusText } }) => {
+        this.setState({
+          httpStatus: status,
+          httpStatusText: statusText
+        });
+      })
   }
 
   render() {
     const {
+      httpStatus,
+      httpStatusText,
       firstName,
       lastName,
       bio,
-      email
-    } = this.state
+      email,
+      pronouns,
+      nationalGoverningBodies,
+      certifications
+    } = this.state;
+
+    if (httpStatus !== 200) {
+      return (
+        <h1>
+          {httpStatusText}
+        </h1>
+      )
+    }
 
     return (
       <div>
-        <h1>Referee Profile</h1>
-        <h3>
-          {`Name: ${firstName} ${lastName}`}
-        </h3>
-        <h3>
-          {`Bio: ${bio}`}
-        </h3>
-        <h3>
-          {`Email: ${email}`}
-        </h3>
+        <h1>
+          {`${firstName} ${lastName}`}
+        </h1>
+        <dl>
+          {pronouns
```

Comment:
Let's include the show_pronouns attribute here and put in a checkbox to allow refs to change that setting.

Reply:
I put that into the RefereeSerializer, because I don’t think that the backend should even send the pronouns to the frondend if the referee chose to not have their pronouns publicly shown. But I’m not sure if thats the right way to do it?

Reply:
In this scenario, I think it's a little easier if the component renders pronouns if at least one of the following are true: 1) The current user that is logged has the same id as the referee being shown 2) The referee data returned includes `show_pronouns` as true. To determine the first condition I can add an attribute in #22 on the referee serializer that is only true if `current_user == referee`.